test/entities/swapV3.integration.test.ts

```diff
@@ -0,0 +1,209 @@
+// pnpm test -- swapV3.integration.test.ts
+import { config } from 'dotenv';
+config();
+import {
+    Address,
+    createTestClient,
+    http,
+    parseEther,
+    publicActions,
+    walletActions,
+    Client,
+    PublicActions,
+    TestActions,
+    WalletActions,
+} from 'viem';
+import { CHAINS, ChainId, SwapKind, Path, Token, Swap } from '../../src';
+import { forkSetup } from '../lib/utils/helper';
+import { ANVIL_NETWORKS, startFork } from '../anvil/anvil-global-setup';
+import { POOLS, TOKENS } from 'test/lib/utils/addresses';
+import { SwapV3 } from '@/entities/swap/swapV3';
+import {
+    assertSwapExactIn,
+    assertSwapExactOut,
+} from 'test/lib/utils/swapHelpers';
+
+const balancerVersion = 3;
+const chainId = ChainId.SEPOLIA;
+// blockNo with guaranteed liquidity
+const blockNo = 5287755n;
+
+const { rpcUrl } = await startFork(ANVIL_NETWORKS.SEPOLIA, undefined, blockNo);
+
+const BAL = TOKENS[chainId].BAL;
+const WETH = TOKENS[chainId].WETH;
+
+describe('SwapV3', () => {
+    let client: Client & PublicActions & TestActions & WalletActions;
+    let testAddress: Address;
+    const pathBalWeth: Path = {
+        balancerVersion: 3,
+        tokens: [
+            {
+                address: TOKENS[chainId].BAL.address,
+                decimals: TOKENS[chainId].BAL.decimals,
+            },
+            {
+                address: TOKENS[chainId].WETH.address,
+                decimals: TOKENS[chainId].WETH.decimals,
+            },
+        ],
+        pools: [POOLS[chainId].MOCK_WEIGHTED_POOL.id],
+        inputAmountRaw: 100000000000n,
+        outputAmountRaw: 100000000000n,
+    };
+
+    beforeAll(async () => {
+        client = createTestClient({
+            mode: 'anvil',
+            chain: CHAINS[chainId],
+            transport: http(rpcUrl),
+        })
+            .extend(publicActions)
+            .extend(walletActions);
+
+        testAddress = (await client.getAddresses())[0];
+    });
+
+    beforeEach(async () => {
+        await forkSetup(
+            client,
+            testAddress,
+            [WETH.address, BAL.address],
+            [WETH.slot as number, BAL.slot as number],
+            [parseEther('100'), parseEther('100')],
+            undefined,
+            balancerVersion,
+        );
+    });
+
+    describe('query method should return correct updated', () => {
+        test('GivenIn', async () => {
+            const swap = new SwapV3({
```

Comment:
Should we instantiate from Swap class instead?
Then we make sure the whole flow is working as expected and avoid the need to import directly from SwapV3?
Or the goal is to actually isolate things while testing?

Reply:
I think at some point I thought it should isolate SwapV3 but as its an integrations test its probably best to test whole flow. Changed here and for V2 test.